cfpb/teachers-digital-platform
# Added Start Page, new content pages and more unit tests

Short description explaining the high-level reason for the pull request

## Additions

- [CRT Start Page](http://localhost:8000/tdp/crt-start) http://localhost:8000/tdp/crt-start
- New content pages 
  - [ ] Content Elementary Criterion Page 
  - [ ] Content Middle Criterion Page 
  - [ ] Content High Criterion Page 
- More unit test
  - [ ] Configured Jest, Enzyme, and Mocking for REACT tests
  - [ ] Updated setup.sh to work with REACT app code build and tests
  - [ ] Have 7 tests right now, but they prove out all the different types of functionality that will need to be tested across all REACT components
- Pulled in most recent STYLES from the following prototypes pages
  - [ ] crt-start
  - [ ] crt-survey
- All 4 distinctives & 3 content distinctive pages are basically duplicated.  Except their values are functioning as expected
- Updated setup.sh & gulp tasks to work with the CRTool
  - added gulp/tasks/crtool.js
  - cleaned up gulp/tasks/test.js : removed code to execute non existant Django tests (will add back if we need it later)

## Testing
- Make sure you run ./setup.sh
- Verify the tdp flag is set to true
- *Start Page*
  - Navigate to http://localhost:8000/tdp/crt-start
  - You will see the START page in the screen shot below
  - Fill out the three values and click "Begin Review"
     - You should redirected to the Survey page
     - Verify the values are in the "Debug Values" section at the bottom of the page
     - NOTE: if you don't see them click the "Start over with a new review" and follow the above steps again.
     - If you navigate back to the  start page the values should be pre-populated (as long as you did not first click "Start over with a new review"
- *Survey Page*
  - Navigate to http://localhost:8000/tdp/crt-survey
  - Click each of the buttons at the top and verify the page below it swaps accordingly
    - The Content Distinctive will have different page depending on the value you choose on the start page.  Elementary, Middle, or High school 
  - Notice the Distinctive button menu is now duplicated at the bottom of the screen (It should always mirror the changes that are reflected in the top menu)
  - Check some of the radio buttons and fill in some of the text 
  - Verify the selections and text are saved while switching pages
  - Even close the web browser and the selections should be saved
  - Any page you start on the button at the top should state 'in progress'
  - If you click the "Start over with a new review" it should clear all selection and text values and redirect you back to the start page
  - Clicking on "Continue to Summary" will mark the current distinctive as complete and it should have a different styling


## Review

- @wpears @Scotchester @willbarton @virtix @dcmouyard @rrstoll 
   - I realize you are all busy, but thought I would tag all of you incase you want to see the improvements from my initial PR.

[Preview this PR without the whitespace changes](?w=0)

## Screenshots


## Notes
- *More unit tests are coming*, Struggled greatly trying to get JEST, Enzyme, Gulp, & react-scripts to work together.  
- Currently have 7 unit tests, but they cover a lot.  Specifically they cover snapshots of the REACT components, and one is a proof of concept on how to verify a Component "OnClick" event is invoked and verified.
  - Going forward I will be taking a TDD approach (now that I have JEST correctly working)

## Checklist
